### PR TITLE
CUMULUS-1560: Update SNS & SQS docs

### PR DESCRIPTION
**Pull request by markdboyd**
- Update SNS docs
- Update SQS docs
- Add `custom_queues` variable to add new queues to workflow message template
- Add module output for `sqs2sfThrottle` Lambda ARN
- Change behavior of `queue_execution_limits` variable integration with workflow message template

### CODE REVIEW

File: docs/data-cookbooks/sns.md [hunk 1]
```diff
@@ -41,54 +57,79 @@ If you are not seeing records persist to the database or show up in the Cumulus
...
-      additionalReceiver:                 # name of the new subscription.
-        endpoint:
-          function: Fn::GetAtt
-          array:
-            - SnsS3TestLambdaFunction     # a lambda configured in lambdas.yml
-            - Arn
-        protocol: lambda
+Additional listeners to SNS topics can be configured in a `.tf` file for your Cumulus deployment. Shown below is configuration that subscribes an additional Lambda function (`test_lambda`) to receive messages from the `report_executions` SNS topic. To subscribe to the `report_granules` or `report_pdrs` SNS topics instead, simply replace `report_executions` in the code block below with either of those values.
+
+```hcl
```
Comment (laurenfrederick) [inline docs/data-cookbooks/sns.md]: This is a nice example

File: docs/data-cookbooks/throttling-queued-executions.md [hunk 2]
```diff
@@ -16,29 +16,70 @@ Limiting the number of executions that can be running from a given queue is usef
...
-      retry: 30
-      maxExecutions: 5
-      consumer:
-        - lambda: sqs2sfThrottle    # you must use this lambda
-          schedule: rate(1 minute)
-          messageLimit: '{{sqs_consumer_rate}'
-          state: ENABLED
+#### Add a new queue
+
+In a `.tf` file for your [Cumulus deployment](./../deployment/README.md#deploy-the-cumulus-instance), add a new SQS queue:
```
Comment (laurenfrederick) [inline docs/data-cookbooks/throttling-queued-executions.md]: This link doesn't work

File: docs/data-cookbooks/throttling-queued-executions.md [hunk 3]
```diff
@@ -16,29 +16,70 @@ Limiting the number of executions that can be running from a given queue is usef
...
+resource "aws_sqs_queue" "background_job_queue" {
+  name                       = "${var.prefix}-backgroundJobQueue"
+  receive_wait_time_seconds  = 20
+  visibility_timeout_seconds = 60
+}
+```
+
+#### Set maximum executions for the queue
+
+Define the `throttled_queues` variable for the `cumulus` module in your [Cumulus deployment](./../deployment/README.md#deploy-the-cumulus-instance) to specify the maximum concurrent executions for the queue.
```
Comment (laurenfrederick) [inline docs/data-cookbooks/throttling-queued-executions.md]: This one needs to get fixed, like above.

File: docs/data-cookbooks/throttling-queued-executions.md [hunk 4]
```diff
@@ -16,29 +16,70 @@ Limiting the number of executions that can be running from a given queue is usef
...
+  function_name = module.cumulus.sqs2sfThrottle_lambda_function_arn
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.background_job_queue_watcher.arn
+}
+```
 
-Once you [re-deploy your Cumulus application](../deployment/deployment-readme#update-cumulus), all of your workflow templates will be updated to the include information about your queue (the output below is partial output from an expected workflow template):
+#### Re-deploy your Cumulus application
+
+Follow the instructions to [re-deploy your Cumulus application](../deployment/upgrade.md#update-cumulus-resources). After you have re-deployed, your workflow template will be updated to the include information about the queue (the output below is partial output from an expected workflow template):
```
Comment (laurenfrederick) [inline docs/data-cookbooks/throttling-queued-executions.md]: Link doesn't work

File: docs/data-cookbooks/throttling-queued-executions.md [hunk 5]
```diff
@@ -55,42 +96,66 @@ Once you [re-deploy your Cumulus application](../deployment/deployment-readme#up
...
-        task_config:
-            # configure the step to use your new queue
-            queueUrl: '{$.meta.queues.backgroundJobQueue}'
-            parsePdrMessageTemplateUri: '{$.meta.templates.ParsePdr}'
-            provider: '{$.meta.provider}'
-            collection: '{$.meta.collection}'
-  ```
-
-After making these changes, re-deploy your Cumulus application for the execution throttling to take effect.
+After making these changes, [re-deploy your Cumulus application](../deployment/upgrade.md#update-cumulus-resources) for the execution throttling to take effect on workflow executions queued by these workflows.
```
Comment (laurenfrederick) [inline docs/data-cookbooks/throttling-queued-executions.md]: Fix this link too

File: docs/data-cookbooks/throttling-queued-executions.md (hunk 2 above)
Comment (laurenfrederick) [reply docs/data-cookbooks/throttling-queued-executions.md]: This one should be this `./../deployment/deployment-readme#deploy-the-cumulus-instance`

File: docs/data-cookbooks/throttling-queued-executions.md (hunk 4 above)
Comment (laurenfrederick) [reply docs/data-cookbooks/throttling-queued-executions.md]: Not sure which page you were trying to go to here, was trying to find the right link

Comment (markdboyd) [reply docs/data-cookbooks/throttling-queued-executions.md]: This text in parentheses `the output below...` is not intended to be a link, just an aside

**Pull request merged**